src/Juvix/Backends/Michelson/Compilation/Util.hs

```diff
@@ -0,0 +1,135 @@
+module Juvix.Backends.Michelson.Compilation.Util where
+
+import Juvix.Backends.Michelson.Compilation.Types
+import Juvix.Library hiding (Type)
+import Michelson.TypeCheck
+import qualified Michelson.Typed as MT
+import Michelson.Untyped
+
+stackToStack ∷ Stack → SomeHST
+stackToStack [] = SomeHST SNil
+stackToStack ((_, ty) : xs) =
+  case stackToStack xs of
+    SomeHST tail →
+      MT.withSomeSingT (MT.fromUType ty) $ \sty →
+        SomeHST (sty -:& tail)
+
+pack ∷
+  ∀ m.
+  (HasThrow "compilationError" CompilationError m) ⇒
+  Type →
+  m ExpandedInstr
+pack (Type TUnit _) = pure (PUSH "" (Type TUnit "") ValueUnit)
+pack ty = throw @"compilationError" (NotYetImplemented ("pack: " <> show ty))
+
+unpack ∷
+  ∀ m.
+  ( HasState "stack" Stack m,
+    HasThrow "compilationError" CompilationError m
+  ) ⇒
+  Type →
+  [Maybe Symbol] →
+  m ExpandedOp
+unpack (Type ty _) binds =
+  case ty of
+    Tbool → do
+      modify @"stack" (drop 1)
+      return (SeqEx [])
+    TPair _ _ fT sT →
+      case binds of
+        [Just fst, Just snd] → do
+          modify @"stack" ((<>) [(VarE fst, fT), (VarE snd, sT)] . drop 1)
```

Comment:
We should make a function to abstract the pattern of `append thing . drop 1`

Reply:
OK, done.